napari/_qt/qt_main_window.py

```diff
@@ -82,9 +85,12 @@ class _QtMainWindow(QMainWindow):
     # *no* active windows, so we want to track the most recently active windows
     _instances: ClassVar[List['_QtMainWindow']] = []
 
-    def __init__(self, viewer: 'Viewer', parent=None) -> None:
+    def __init__(
+        self, viewer: 'Viewer', window: 'Window', parent=None
+    ) -> None:
         super().__init__(parent)
         self._ev = None
+        self._window = window
```

Comment:
Should we add a comment about injection here? Also related comment: https://github.com/napari/napari/pull/4826/files#r942758421

Reply:
Done